src/ssh-compute.ts

```diff
@@ -0,0 +1,189 @@
+    const ssh_public_key = core.getInput('ssh_public_key');
+    const ssh_private_key = core.getInput('ssh_private_key');
+    const ssh_keys_folder = core.getInput('ssh_keys_folder');
+    const container = core.getInput('container');
+    const sshArgs = core.getInput('ssh_args');
+    let command = core.getInput('command');
+    const script = core.getInput('script');
+    const projectId = core.getInput('project_id');
+    let gcloudVersion = core.getInput('gcloud_version');
+
+    // Flags
+    const flags = core.getInput('flags');
+    const installBeta = true; // Flag for installing gcloud beta components
+    let cmd;
+
+    if (command && script) {
+      throw new Error('Both `command` and `entrypoint` inputs set - Please select one.');
+    }
+
+    if (user) {
+      instanceName = `${user}@${instanceName}`;
+    }
+
+    cmd = [
+      'compute',
+      'ssh',
+      instanceName,
+      '--zone',
+      zone,
+      '--ssh-key-file',
+      `${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`,
+      '--quiet', // we need to ignore prompts from console
+      '--tunnel-through-iap',
+    ];
+
+    // Save public and private ssh keys to the temp folder
+    await fs.mkdir(ssh_keys_folder, { recursive: true });
+    await fs.writeFile(`${ssh_keys_folder}/${SSH_PUBLIC_KEY_FILENAME}`, ssh_public_key, {
+      mode: 0o600,
+    });
+    await fs.writeFile(`${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`, '', { mode: 0o600 });
+    for (const key of ssh_private_key.split(/(?=-----BEGIN)/)) {
+      await fs.appendFile(`${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`, key.trim() + '\n');
+    }
+
+    if (container) {
+      cmd.push('--container', container);
+    }
+
+    if (flags) {
+      const flagList = parseFlags(flags);
+      if (flagList) cmd = cmd.concat(flagList);
+    }
+
+    if (script) {
+      try {
+        await fs.access(script);
+      } catch (error: any) {
+        const message = 'Script can not be found. Check script input path.';
+        throw new Error(message);
+      }
+
+      const commandData = (await fs.readFile(script)).toString('utf8');
+      command = `bash -c \"${commandData}\"`; // eslint-disable-line no-useless-escape
+    }
+
+    if (sshArgs) {
+      cmd.push(`-- ${sshArgs}`);
+    }
+
+    // Install gcloud if not already installed.
+    if (!gcloudVersion || gcloudVersion == 'latest') {
+      gcloudVersion = await setupGcloud.getLatestGcloudSDKVersion();
+    }
+    if (!setupGcloud.isInstalled(gcloudVersion)) {
+      await setupGcloud.installGcloudSDK(gcloudVersion);
+    } else {
+      const toolPath = toolCache.find('gcloud', gcloudVersion);
+      core.addPath(path.join(toolPath, 'bin'));
+    }
+
+    // Authenticate gcloud SDK.
+    await setupGcloud.authenticateGcloudSDK();
+    const authenticated = await setupGcloud.isAuthenticated();
+    if (!authenticated) {
+      throw new Error('Error authenticating the Cloud SDK.');
+    }
+
+    // set PROJECT ID
```

Comment:
Since this is a new action, I think we can simplify it, but the project_id should be set by `auth`. You only need to append `--project` if the user provided an explicit project parameter: https://github.com/google-github-actions/deploy-cloudrun/blob/c5e1552096165af5a328a6b3a4a28a733ea9d072/src/deploy-cloudrun.ts#L197-L206